README.md

```diff
@@ -1 +1,388 @@
-# backend-palette-picker
+# Palette Picker API
+
+## About
+
+Palette Picker is a website that generates random color combinations (palettes) of 5 colors. The API allows for saving palettes in folders. 
+
+The available endpoints are: 
+
+##### Folders
+
+* GET /api/v1/folders to get all folders in the database
+* GET /api/v1/folders?name=[name] to get a specific folder by name
+* GET /api/v1/folders/:id to get a specific folder by id
+* POST /api/v1/folders to post a new folder to the database
+* PATCH /api/v1/folders/:id to edit an existing folder by id
+* DELETE /api/v1/folders/:id to delete a folder and it's associated palettes from the database
+
+##### Palettes
+
+* GET /api/v1/palettes to get all palettes in the database
+* GET /api/v1/palettes/:id to get a specific palette by id
+* POST /api/v1/palettes to post a new palette to the database
+* PATCH /api/v1/palettes/:id to edit an existing palette by id
+* DELETE /api/v1/palettes/:id to delete a palette from the database
+
+## Endpoints
+
+#### List all folders
+
+```
+GET /api/v1/folders
+```
+
+
+##### Parameters
+
+| Name | Type | Description | 
+| ---  | ---- | ----------- |
+| `id` | `number` | Unique id of folder |
+| `name` | `string` | User-created name of folder |
+
+##### Response
+
+Returns an array containing all folders in database.
+
+` Status: 200 OK `
+
+```json
+[
+    {
+        "id": 3,
+        "name": "Test Folder 2",
+        "created_at": "2019-10-09T01:35:09.554Z",
+        "updated_at": "2019-10-09T01:35:09.554Z"
+    },
+    {
+        "id": 4,
+        "name": "Blah",
+        "created_at": "2019-10-10T19:09:12.843Z",
+        "updated_at": "2019-10-10T19:09:12.843Z"
+    }
+]
+
+```
+
+
+
+
+
+---
+#### Find specific folder by id
+
+```
+GET /api/v1/folders/:id
+```
+
+##### Parameters
+
+| Name | Type | Description | 
+| ---  | ---- | ----------- |
+| `id` | `number` | Unique id of folder |
+| `name` | `string` | User-created name of folder |
+
+##### Response
+
+Returns an array containing folder matching the `id` parameter
+
+```json
+[
+    {
+        "id": 3,
+        "name": "Test Folder 2",
+        "created_at": "2019-10-09T01:35:09.554Z",
+        "updated_at": "2019-10-09T01:35:09.554Z"
+    }
+]
+
+```
```

Comment:
These look excellent, but do we want to add error responses as well so that developers know what an error will be